cdap-ui/app/cdap/components/CaskWizards/AddNamespace/index.js

```
@@ -36,8 +37,14 @@ export default class AddNamespaceWizard extends Component {
   createNamespace() {
     return PublishNamespace()
       .flatMap(
-        () => {
-          return PublishPreferences();
+        (res) => {
+          if (res.includes('already exists')) {
+            return Rx.Observable.create((observer) => {
```

I think if you just need to throw an error, you can use `Rx.Observable.throw`.

https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/throw.md

---

I tried doing that, but other functions that inherit this one as a prop (e.g. `submitForm` in the `Wizard` component) expect the return value to be an observable, so it didn't work.